Source/WTF/wtf/unix/UnixFileDescriptor.h

```diff
@@ -55,6 +55,12 @@ class UnixFileDescriptor {
         m_value = o.release();
     }
 
+    explicit UnixFileDescriptor(const UnixFileDescriptor& o)
+    {
+        if (o.m_value >= 0)
+            m_value = dupCloseOnExec(o.m_value);
+    }
```

Comment:
It doesn't make sense for a noncopyable object to have a copy constructor. You _could_ remove the `WTF_MAKE_NONCOPYABLE`, but I don't think it's a good idea. It's safer to force programmers to be explicit about duping file descriptors.

I would give SharedMemoryHandle a non-default copy constructor that manually calls UnixFileDescriptor::duplicate instead.

Reply:
(The reason I suggested this was to prevent accidental dup operations. It's not the end of the world since it won't cause file descriptors to be leaked, but it's still not great.)

Reply:
I agree that making accidentally copies harder is worthwhile.

I went for the `explicit` copy constructor, since it was what was agreed upon for `MachSendRight`, and keeping a common interface is helpful (since they're used interchangeably with preprocessor macros).

Reply:
Hm, OK I see. Actually I didn't know that was possible. That's probably fine.

Reply:
It was new to me too :)